### ISSUE TITLE
히든 레이어에 대한 질문

### ISSUE THREAD

**Issue by L-cloud** [codingeverybody/codingyahac#1021]
# 해결하고자 하는 문제

텐서 플로우 영상을 보며 공부하던 중 조금 이해가 안되는 부분이 생겼습니다.
히든레이어에 대한 질문인데요
음 여러 변수들의 관계를 찾아서 어떤 값을 예측한다는 것이라고 머신 러닝을 이해했습니다.
밑의 사례는 보스턴 집값을 13가지 변수 사이의 관계를 살펴보며 집값을 예측하는 모델이잖아요.
그런데 히든레이어가 들어옵니다.. 그냥 단순히 
H = tf.keras.layers.Dense(10, activation='swish')(X) 이 한줄 코드로
13가지 변수들의 관계를 통해 집값이라는 1개의 값이 아닌
중간에 또 10개의 변수를 만들어 내는 건가요??
그래서 그 10개의 변수가 1개의 값인 집값을 예측 하고요...
의문은 변수가 다시 또 변수를 만들어 내는게 맞는가 하는 점입니다..
현실로 예를 들면
집값 요인이 crim', 'zn', 'indus', 'chas', 'nox', 이 4개만 있다고 생각하고
위치, 평수, 층, 소음 이렇게 4개 변수인데
여기서 히든 레이어를 첨가해서 2개의 변수로 만든다고 한다면
청결도, 창문갯수 등 이런 새로운 변수 2개를 만들어 내는건가요,,?
H = tf.keras.layers.Dense(2, activation='swish')(X) 뭐 이렇게 바꾸면..? 

# 코드 혹은 오류
```
import tensorflow as tf
import pandas as pd
파일경로 = 'https://raw.githubusercontent.com/blackdew/tensorflow1/master/csv/boston.csv'
보스턴 = pd.read_csv(파일경로)

# 종속변수, 독립변수
독립 = 보스턴[['crim', 'zn', 'indus', 'chas', 'nox', 
            'rm', 'age', 'dis', 'rad', 'tax',
            'ptratio', 'b', 'lstat']]
종속 = 보스턴[['medv']]
print(독립.shape, 종속.shape)
X = tf.keras.layers.Input(shape=[13])
H = tf.keras.layers.Dense(10, activation='swish')(X)
Y = tf.keras.layers.Dense(1)(H)
model = tf.keras.models.Model(X, Y)
model.compile(loss='mse')
model.summary()
model.fit(독립, 종속, epochs=100)


```

**Comment by jiyongjung0** [comment]
네. 아주 핵심적인 질문입니다. 말씀하신대로 사람들은 늘 이유가 있으면 결과가 있고, 변수로 다른 변수를 계산하는 것을 좋아했지요. 그래서 어떤 변수를 만들어야 원하는 척도를 잘 표현할 수 있을까 늘 고민이 많았답니다. 하지만 딥러닝에서는, 사람이 적절히 학습할 내용만 정해주면 기계가 알아서 그런 잘 되는 변수(?)를 알아서 찾아낸다는 것이 핵심입니다. 입력과 출력 데이터만 왕창 주면, 그 사이에 어떤 변수를 만드는 것이 좋은지 알아서 학습해서 알아서 결과를 내놓는다고 생각할 수도 있습니다.

실제로 딥러닝 학습 결과를 다시 분석해서 어떤 요소가 중요한지 거꾸로 연구하기도 합니다.

집값을 잘 못 예측하는 것 같으면 히든레이어를 두 개도 넣어보고 10개도 넣어볼 수도 있지요. 그러다보면 신기하게 컴퓨터가 학습을 잘 하는 것이 딥러닝입니다. 😎
